### PR TITLE
Bigtable read and filter snippets

### PR DESCRIPTION
**Pull request by billyjacobson**

### CODE REVIEW

**Comment by billyjacobson** [comment]
I wasn't able to figure out why the tests are failing in kokoro. I ran them on my machine using the same project the tests run on, but they worked.

File: bigtable/src/filter_snippets.php [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+    'filter_limit_value_range',
+    'filter_limit_value_regex',
+    'filter_limit_timestamp_range',
+    'filter_limit_block_all',
+    'filter_limit_pass_all',
+    'filter_modify_strip_value',
+    'filter_modify_apply_label',
+    'filter_composing_chain',
+    'filter_composing_interleave',
+    'filter_composing_condition'];
```
Comment (bshaffer) [inline bigtable/src/filter_snippets.php]: nit: put closing `];` on a new line (with no indent)

File: bigtable/src/filter_snippets.php [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+                $labels =
+                    $data[$i]['labels'] == "" ? "" : sprintf(" [%s]", $data[$i]['labels']);
+
+                printf(
+                    "\t%s: %s @%s%s" . PHP_EOL,
+                    $col,
+                    $data[$i]['value'],
+                    $data[$i]['timeStamp'],
+                    $labels
+                );
```
Comment (bshaffer) [inline bigtable/src/filter_snippets.php]: This doesn't look great. Can we do this instead? 
```php
                printf(
                    "\t%s: %s @%s%s" . PHP_EOL,
                    $col,
                    $data[$i]['value'],
                    $data[$i]['timeStamp'],
                    $data[$i]['labels'] ? sprintf(" [%s]", $data[$i]['labels']) : ''
                );

File: bigtable/src/read_snippets.php [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+                $labels =
+                    $data[$i]['labels'] == "" ? "" : sprintf(" [%s]", $data[$i]['labels']);
+
+                printf(
+                    "\t%s: %s @%s%s" . PHP_EOL,
+                    $col,
+                    $data[$i]['value'],
+                    $data[$i]['timeStamp'],
+                    $labels
+                );
```
Comment (bshaffer) [inline bigtable/src/read_snippets.php]: Same here

File: bigtable/src/writes/write_batch.php [hunk 4]
```diff
@@ -56,7 +56,8 @@
         ->upsert($columnFamilyId, "connected_wifi", 1, $timestamp)
         ->upsert($columnFamilyId, "os_build", "12145.0.0-rc6", $timestamp)];
 
-$table->mutateRows([
+$table->mutateRows(
+    [
         "tablet#a0b81f74#20190501" => $mutations[0],
         "tablet#a0b81f74#20190502" => $mutations[1]]
 );
```
Comment (bshaffer) [inline bigtable/src/writes/write_batch.php]: This should look like:

```php
$table->mutateRows([
    "tablet#a0b81f74#20190501" => $mutations[0],
    "tablet#a0b81f74#20190502" => $mutations[1],
]);
```

File: bigtable/src/writes/write_increment.php [hunk 5]
```diff
@@ -46,7 +46,7 @@
 ]);
 $table = $dataClient->table($instance_id, $table_id);
 
-$timestamp = time() * 1000;
+$timestamp = time() * 1000 * 1000;
```
Comment (bshaffer) [inline bigtable/src/writes/write_increment.php]: Let's make this `$timestampMicros` for clarity

File: bigtable/test/filterTest.php [hunk 6]
```diff
@@ -0,0 +1,860 @@
...
+            self::$projectId,
+            self::$instanceId,
+            self::$tableId,
+            "filter_limit_row_sample"
+        ]);
+        $result = "Reading data for row ";
+        $this->assertContains($result, trim($output));
+    }
+
+
```
Comment (bshaffer) [inline bigtable/test/filterTest.php]: unnecessary newline

File: bigtable/test/filterTest.php [hunk 7]
```diff
@@ -0,0 +1,860 @@
...
+        $this->useResourceExhaustedBackoff();
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        $instanceName = self::$bigtableInstanceAdminClient->instanceName(self::$projectId, self::$instanceId);
+        self::$bigtableInstanceAdminClient->deleteInstance($instanceName);
+    }
+
+
```
Comment (bshaffer) [inline bigtable/test/filterTest.php]: unnecessary newline

File: bigtable/test/filterTest.php [hunk 8]
```diff
@@ -0,0 +1,860 @@
...
+use Google\Cloud\Bigtable\Admin\V2\BigtableInstanceAdminClient;
+use Google\Cloud\Bigtable\Admin\V2\ColumnFamily;
+use Google\Cloud\Bigtable\BigtableClient;
+use Google\Cloud\Bigtable\Mutations;
+use PHPUnit\Framework\TestCase;
+
+use Google\Cloud\Bigtable\Admin\V2\BigtableTableAdminClient;
+use Google\Cloud\Bigtable\Admin\V2\Table;
+use Google\Cloud\TestUtils\TestTrait;
+use Google\Cloud\TestUtils\ExponentialBackoffTrait;
```
Comment (bshaffer) [inline bigtable/test/filterTest.php]: Nit but let's make these alphabetical order with no newline break in between: 

```php
use Google\Cloud\Bigtable\Admin\V2\BigtableInstanceAdminClient;
use Google\Cloud\Bigtable\Admin\V2\BigtableTableAdminClient;
use Google\Cloud\Bigtable\Admin\V2\ColumnFamily;
use Google\Cloud\Bigtable\Admin\V2\Table;
use Google\Cloud\Bigtable\BigtableClient;
use Google\Cloud\Bigtable\Mutations;
use Google\Cloud\TestUtils\ExponentialBackoffTrait;
use Google\Cloud\TestUtils\TestTrait;
use PHPUnit\Framework\TestCase;
```

File: bigtable/test/filterTest.php [hunk 9]
```diff
@@ -0,0 +1,860 @@
...
+
+final class FilterTest extends TestCase
+{
+    use TestTrait;
+    use ExponentialBackoffTrait;
+
+    const INSTANCE_ID_PREFIX = 'phpunit-test-';
+    const TABLE_ID_PREFIX = 'mobile-time-series-';
+    private static $bigtableInstanceAdminClient;
```
Comment (bshaffer) [inline bigtable/test/filterTest.php]: add newline between `const` and `private`

File: bigtable/test/filterTest.php [hunk 10]
```diff
@@ -0,0 +1,860 @@
...
+	data_plan_10gb: 1 @%1$s
+Column Family stats_summary
+	connected_cell: 1 @%1$s
+	connected_wifi: 1 @%1$s
+	os_build: PQ2A.190401.002 @%1$s', self::$timestamp, self::$timestamp_minus_hr);
+
+        $this->assertEquals($result, trim($output));
+    }
+
+
```
Comment (bshaffer) [inline bigtable/test/filterTest.php]: unnecessary newline, and please remove for all proceeding occurrences as well

File: bigtable/test/readTest.php [hunk 11]
```diff
@@ -0,0 +1,337 @@
...
+use Google\Cloud\Bigtable\Admin\V2\BigtableInstanceAdminClient;
+use Google\Cloud\Bigtable\Admin\V2\ColumnFamily;
+use Google\Cloud\Bigtable\BigtableClient;
+use Google\Cloud\Bigtable\Mutations;
+use PHPUnit\Framework\TestCase;
+
+use Google\Cloud\Bigtable\Admin\V2\BigtableTableAdminClient;
+use Google\Cloud\Bigtable\Admin\V2\Table;
+use Google\Cloud\TestUtils\TestTrait;
+use Google\Cloud\TestUtils\ExponentialBackoffTrait;
```
Comment (bshaffer) [inline bigtable/test/readTest.php]: order alphabetically

**Pull request merged**